### PR TITLE
Adds model convertors to deal with pyodide issues

### PR DESCRIPTION
**Pull request by stuartlynn**
Adds a pooch processor to perform the necessary corrections to the HistGradientBoosting models to allow then to work in pyodide.

These should only run in that environment.

### CODE REVIEW

File: demoland_engine/data.py [hunk 1]
```diff
@@ -38,3 +39,46 @@
...
+  # Only used if is_categorical is True
+  ('bitset_idx', np.uint32)
+])
+
+# pooch processor to fix pyodide bug
+def pyodide_convertor(fname, action, pup):
+    try:
+        # Check to see if we are in a pyodide environment
+        import pyodide_js
+        model = joblib.load(fname)
```
Comment (martinfleis) [inline demoland_engine/data.py:139]: It probably won't work with current models (old Newcastle) as those are pickles and not joblib files.

Comment (stuartlynn) [reply demoland_engine/data.py:139]: As suggested, I will add a file prefix check and use the relevant importer

Comment (martinfleis) [reply demoland_engine/data.py:139]: updated on main, so joblib should be enough

File: demoland_engine/data.py [hunk 2]
```diff
@@ -38,3 +39,46 @@
...
+        model._predictors[i][0].nodes = model._predictors[i][0].nodes.astype(PREDICTOR_RECORD_DTYPE_2)
```
Comment (lesteve) [inline demoland_engine/data.py]: Using `casting="same_kind"` is probably a good idea (`casting="unsafe"` by default), to fail early if for some reason the dtypes are not compatible:
```suggestion
        model._predictors[i][0].nodes = model._predictors[i][0].nodes.astype(PREDICTOR_RECORD_DTYPE_2, casting="same_kind")
```

Comment (stuartlynn) [reply demoland_engine/data.py]: Thanks for the suggestion!

**Comment by martinfleis** [comment]
@stuartlynn if I resolve merge conflicts, is this mergeable?

**Comment by stuartlynn** [comment]
Hey yup I think this should be good to go

File: demoland_engine/data.py [hunk 3]
```diff
@@ -98,6 +99,55 @@
...
+
+
+# pooch processor to fix pyodide bug
+def pyodide_convertor(fname, action, pup):
+    try:
+        # Check to see if we are in a pyodide environment
+        import pyodide_js
+
+        model = joblib.load(fname)
+        model._predictors[i][0].nodes = model._predictors[i][0].nodes.astype(
```
Comment (martinfleis) [inline demoland_engine/data.py]: The `i` here is not defined. Should there be some form of a loop here?

Comment (stuartlynn) [reply demoland_engine/data.py]: Yeah sorry must have missed it porting the code over. Needs to iterate over the _perdictors 

so 
```python
for i,_ in enumerate(model._predictors):
        model._predictors[i][0].nodes = model._predictors[i][0].nodes.astype(
```

**Pull request merged**